spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreOperations.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core;
+
+/**
+ * An interface of operations that can be done with Cloud Datastore.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+public interface DatastoreOperations {
+
+	/**
+	 * Get an entity based on a id.
+	 * @param id the id of the entity
```

Comment:
We probably need to document the possible options for the id: `String`, `Long`, `Key`?

Reply:
@meltsufin I'll make this change.

This PR is already nearly 100 lines. We would be at ~900 lines with tests. Removing the options class only saves like 50 lines. 

Please let's stick with 100 line for now.